src/ingress_definition.py

```diff
@@ -316,16 +316,54 @@ def proxy_connect_timeout(self) -> str:
 
     @property
     def rewrite_enabled(self) -> bool:
-        """Return whether rewriting should be enabled from config or relation."""
-        value = self._get_config_or_relation_data("rewrite-enabled", True)
+        """
+        Return whether rewriting should be enabled from config, per-app ingress
+        relation (strip_prefix), or nginx-route relation, in that order.
+        """
+        value = self._get_config("rewrite-enabled")
         # config data is typed, relation data is a string
         # Convert to string, then compare to a known value.
-        return str(value).lower() == "true"
+        if value is not None:
+            return str(value).lower() == "true"
```

Comment:
The default configuration value for the `rewrite-enabled` charm configuration is `false`, which means that `value is not None` will never be `True`.

One solution might be to remove the default value for `rewrite-enabled` in `charmcraft.yaml` so it defaults to `None`.